docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-cloud-providers/azure.md

```diff
@@ -110,6 +124,389 @@ Only hosts expected to be load balancer back ends need to be in this group.
+  name: system:azure-cloud-provider-secret-getter
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:azure-cloud-provider-secret-getter
+  subjects:
+    - kind: ServiceAccount
+      name: azure-cloud-config
+      namespace: kube-system
+  ```
+
+4. Click the **Create** button to submit the form and create the cluster.
+
+</TabItem>
+
+<TabItem value="RKE1">
+
+1. Choose "External" from the Cloud Provider drop-down in the **Cluster Options** section. This sets `--cloud-provider=external` for Kubernetes components.
+
+2. Install the `cloud-provider-azure` chart after the cluster finishes provisioning. Note that the cluster isn't successfully provisioned and nodes are still in an `uninitialized` state until you deploy the cloud controller manager. This can be done [manually using CLI](#helm-chart-installation-from-cli), or via [Helm charts in UI](#helm-chart-installation-from-ui). 
+
+Refer to the official Azure upstream documentation for the [cloud provider](https://cloud-provider-azure.sigs.k8s.io/install/azure-ccm/).
+
+</TabItem>
+</Tabs>
+
+### Helm Chart Installation from CLI
+
+Official upstream docs for [Helm chart installation](https://github.com/kubernetes/cloud-provider-aws/tree/master/charts/aws-cloud-controller-manager) can be found on Github.
+
+1. Create a `azure-cloud-config` secret with the required [cloud provider config](#cloud-provider-configuration). 
+
+```shell 
+kubectl apply -f azure-cloud-config.yaml
+```
+
+2. Add the Helm repository:
+
+```shell
+helm repo add azure-cloud-controller-manager https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/repo
+helm repo update
+```
+
+3. Create a `values.yaml` file with the following contents to override the default `values.yaml`:
+
+<Tabs groupId="k8s-distro">
+<TabItem value="RKE2">
+
+```yaml
+# values.yaml
+infra:
+  clusterName: <cluster-name>
+cloudControllerManager:
+  cloudConfigSecretName: azure-cloud-config
+  cloudConfig: null
+  clusterCIDR: null
+  enableDynamicReloading: 'true'
+  configureCloudRoutes: 'false'
+  allocateNodeCidrs: 'false'
+  caCertDir: /etc/ssl
+  enabled: true
+  replicas: 1
+  hostNetworking: true
+  nodeSelector:
+    node-role.kubernetes.io/control-plane: 'true'
+  tolerations:
+    - effect: NoSchedule
+      key: node-role.kubernetes.io/master
+    - effect: NoSchedule
+      key: node-role.kubernetes.io/control-plane
+      value: 'true'
+    - effect: NoSchedule
+      key: node.cloudprovider.kubernetes.io/uninitialized
+      value: 'true'
+```
+
+</TabItem>
+
+<TabItem value="RKE">
+
+```yaml
+# values.yaml
+cloudControllerManager:
+  cloudConfigSecretName: azure-cloud-config
+  cloudConfig: null
+  clusterCIDR: null  
+  enableDynamicReloading: 'true'
+  configureCloudRoutes: 'false'
+  allocateNodeCidrs: 'false'
+  caCertDir: /etc/ssl
+  enabled: true
+  replicas: 1
+  hostNetworking: true
+  nodeSelector:
+    node-role.kubernetes.io/controlplane: 'true'
```

Comment:
```
nodeSelector:
  node-role.kubernetes.io/controlplane: 'true'
```
results in failure to deploy `cloud-controller-manager` pod. After changing it to `node-role.kubernetes.io/control-plane: 'true'` the pod gets deployed successfully.

Reply:
Synced up with @vivek-shilimkar offline, we do not need to update the label as RKE1 clusters label nodes as `node-role.kubernetes.io/controlplane: 'true'`. But since helm CLI adds the default `node-role.kubernetes.io/control-plane: ''` anyway, the pods end up not getting deployed. To fix this, we need to set this key as `null` in values.yaml. I have updated the docs to reflect the workaround.